app/models/beta/search/goods_nomenclature_query.rb

```diff
@@ -22,20 +26,22 @@ class GoodsNomenclatureQuery
         'goods_nomenclature_item_id',
       ].freeze
 
-      def self.build(search_query_parser_result)
+      def self.build(search_query_parser_result, filters = [])
         query = new
 
         query.nouns = search_query_parser_result.nouns
         query.noun_chunks = search_query_parser_result.noun_chunks
         query.verbs = search_query_parser_result.verbs
         query.adjectives = search_query_parser_result.adjectives
+        query.filters = filters
 
         query
       end
 
       def query
         candidate_query = { query: { bool: {} } }
 
+        candidate_query[:query][:bool][:filter] = filter_part if filters.any?
         candidate_query[:query][:bool][:must] = must_part if must_part.any?
         candidate_query[:query][:bool][:should] = should_part if should_part.any?
```

Comment:
do we need the ifs here?
maybe it's fine to have 

`candidate_query[:query][:bool][:filter] = nil`

same for must and should

Reply:
This is actually something required by the ES api